src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -24,6 +24,9 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     internal sealed class PEModule : IDisposable
     {
+        // We need to store reference for to keep the metadata alive while symbols have reference to PEModule.
+        private readonly ModuleMetadata _owner;
```

Comment:
I think it's likely always been a problem but we don't work too much with standalone .netmodules so we haven't hit it. It's be good to add a dedicated test that creates a netmodule, only holds on the corresponding PE symbol (and not the metadata) calls GC.Collect and then accesses some members of the symbol.